### PR TITLE
do not allow to set Request.ID different than Operation.ID

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
1. Make Request.ID a proxy for Context.Operation.ID
2. Add CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Changelog
+
+This changelog will be used to generate documentation on [release notes page](http://azure.microsoft.com/en-us/documentation/articles/app-insights-release-notes-dotnet/).
+
+## Version 1.2
+
+- TrackDependency will produce valid JSON when not all required fields were specified.
+- Redundant property ```RequestTelemetry.ID``` is now just a proxy for ```RequestTelemetry.Operation.Id```.
```
Comment (abaranch) [inline CHANGELOG.md]: Will your changes be part of 1.3?

Comment (SergeyKanzhelev) [inline CHANGELOG.md]: I should have spelled it out as vNext so we can put the version later. Will fix it

File: src/Core/Managed/Core.Net46/Extensibility/Implementation/External/OperationContextData.cs [hunk 2]
```diff
@@ -57,38 +57,17 @@ public string Name
...
-            get { return this.tags.GetTagValueOrNull(ContextTagKeys.Keys.OperationParentId); }
-            set { this.tags.SetStringValueOrRemove(ContextTagKeys.Keys.OperationParentId, value); }
-        }
-
-        public string RootId
-        {
-            get { return this.tags.GetTagValueOrNull(ContextTagKeys.Keys.OperationRootId); }
-            set { this.tags.SetStringValueOrRemove(ContextTagKeys.Keys.OperationRootId, value); }
-        }
-
```
Comment (JakubOleksy) [inline src/Core/Managed/Core.Net46/Extensibility/Implementation/External/OperationContextData.cs]: These are breaking changes...so only 2.0?

Comment (SergeyKanzhelev) [inline src/Core/Managed/Core.Net46/Extensibility/Implementation/External/OperationContextData.cs]: these properties were not public. I reverted this change anyway as those files are autogenerated.

Comment (JakubOleksy) [inline src/Core/Managed/Core.Net46/Extensibility/Implementation/External/OperationContextData.cs]: I see. So the public contract did not change? We can change it, but if we do, we need to go to 2.0.

**Comment by abaranch** [comment]
:shipit:

**Comment by JakubOleksy** [comment]
:shipit:

**Pull request merged**